# Mod 2 Review

In [81]:
import pandas as pd
from sklearn.preprocessing import StandardScaler
from statsmodels.formula.api import ols
from scipy import stats
import statsmodels.api as sm
import statsmodels.formula.api as smf
from statsmodels.stats import weightstats
pd.set_option('display.max_columns', 100)

### Check Your Data … Quickly
The first thing you want to do when you get a new dataset, is to quickly to verify the contents with the .head() method.

In [2]:
df = pd.read_csv('movie_metadata.csv')
print(df.shape)
df.head()


(5043, 28)


color      director_name  num_critic_for_reviews  duration  \
0  Color      James Cameron                   723.0     178.0   
1  Color     Gore Verbinski                   302.0     169.0   
2  Color         Sam Mendes                   602.0     148.0   
3  Color  Christopher Nolan                   813.0     164.0   
4    NaN        Doug Walker                     NaN       NaN   

   director_facebook_likes  actor_3_facebook_likes      actor_2_name  \
0                      0.0                   855.0  Joel David Moore   
1                    563.0                  1000.0     Orlando Bloom   
2                      0.0                   161.0      Rory Kinnear   
3                  22000.0                 23000.0    Christian Bale   
4                    131.0                     NaN        Rob Walker   

   actor_1_facebook_likes        gross                           genres  \
0                  1000.0  760505847.0  Action|Adventure|Fantasy|Sci-Fi   
1                 40000.0  309404152.0         Action|Adventure|Fantasy   
2                 11000.0  200074175.0        Action|Adventure|Thriller   
3                 27000.0  448130642.0                  Action|Thriller   
4                   131.0          NaN                      Documentary   

      actor_1_name                                        movie_title  \
0      CCH Pounder                                            Avatar    
1      Johnny Depp          Pirates of the Caribbean: At World's End    
2  Christoph Waltz                                           Spectre    
3        Tom Hardy                             The Dark Knight Rises    
4      Doug Walker  Star Wars: Episode VII - The Force Awakens    ...   

   num_voted_users  cast_total_facebook_likes          actor_3_name  \
0           886204                       4834             Wes Studi   
1           471220                      48350        Jack Davenport   
2           275868                      11700      Stephanie Sigman   
3          1144337                     106759  Joseph Gordon-Levitt   
4                8                        143                   NaN   

   facenumber_in_poster                                      plot_keywords  \
0                   0.0             avatar|future|marine|native|paraplegic   
1                   0.0  goddess|marriage ceremony|marriage proposal|pi...   
2                   1.0                bomb|espionage|sequel|spy|terrorist   
3                   0.0  deception|imprisonment|lawlessness|police offi...   
4                   0.0                                                NaN   

                                     movie_imdb_link  num_user_for_reviews  \
0  http://www.imdb.com/title/tt0499549/?ref_=fn_t...                3054.0   
1  http://www.imdb.com/title/tt0449088/?ref_=fn_t...                1238.0   
2  http://www.imdb.com/title/tt2379713/?ref_=fn_t...                 994.0   
3  http://www.imdb.com/title/tt1345836/?ref_=fn_t...                2701.0   
4  http://www.imdb.com/title/tt5289954/?ref_=fn_t...                   NaN   

  language country content_rating       budget  title_year  \
0  English     USA          PG-13  237000000.0      2009.0   
1  English     USA          PG-13  300000000.0      2007.0   
2  English      UK          PG-13  245000000.0      2015.0   
3  English     USA          PG-13  250000000.0      2012.0   
4      NaN     NaN            NaN          NaN         NaN   

   actor_2_facebook_likes  imdb_score  aspect_ratio  movie_facebook_likes  
0                   936.0         7.9          1.78                 33000  
1                  5000.0         7.1          2.35                     0  
2                   393.0         6.8          2.35                 85000  
3                 23000.0         8.5          2.35                164000  
4                    12.0         7.1           NaN                     0

## Question 1
A Hollywood executive whats to know how much a R-Rated movie released after 2000 will earn. The data above is a sample of some of the movies with that ranting during that time frame, as well as other movies. How would you go about answering her question? Talk through it theoretically and then do it in code.

In [3]:
print(df.keys())
df_r= df[(df['title_year'] > 2000) & (df['content_rating'] == 'R')]['gross']
df_r

Index(['color', 'director_name', 'num_critic_for_reviews', 'duration',
       'director_facebook_likes', 'actor_3_facebook_likes', 'actor_2_name',
       'actor_1_facebook_likes', 'gross', 'genres', 'actor_1_name',
       'movie_title', 'num_voted_users', 'cast_total_facebook_likes',
       'actor_3_name', 'facenumber_in_poster', 'plot_keywords',
       'movie_imdb_link', 'num_user_for_reviews', 'language', 'country',
       'content_rating', 'budget', 'title_year', 'actor_2_facebook_likes',
       'imdb_score', 'aspect_ratio', 'movie_facebook_likes'],
      dtype='object')


84              NaN
94      150350192.0
113      34293771.0
124     139259759.0
126     281492479.0
           ...     
5012     10499968.0
5014            NaN
5019            NaN
5021       192467.0
5026       136007.0
Name: gross, Length: 1390, dtype: float64

In [4]:
df_r.isna().value_counts()

False    1208
True      182
Name: gross, dtype: int64

In [5]:
df_r.dropna(inplace= True)

/Users/avidanberman/opt/anaconda3/lib/python3.7/site-packages/pandas/core/series.py:4494: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self._update_inplace(result)


In [6]:
df_r.isna().value_counts()

False    1208
Name: gross, dtype: int64

In [16]:
print('Mean:', df_r.mean(),'Confidence Interval:', stats.norm.interval(alpha = .95, loc = df_r.mean(), scale = len(df_r)))

Mean: 27648848.437913906 Confidence Interval: (27646480.80142058, 27651216.07440723)


## Question 2

Your ability to answer the first question has the executive excited and now she has many other questions about the types of movies being made and the differences in those movies budgets and gross amounts.  Read through the questions below and determine what type of statistical test you should use for each question and write down the null and alternative hypothesis for those tests.    

- Is there a relationship between the number of facebook likes for a cast and the box office gross of the movie?
- Do foreign films perform differently at the box office than non-foreign films?
- Of all movies created 40% are rated R.
- Is there a relationship between the language of a film and the rating of that film?
- Is there a relationship between the content rating of a film and its budget? 

1)
h0) fbl != gross h1) fbl = gross

Answer: Correlation Coefficient

In [21]:
df[['gross', 'cast_total_facebook_likes']].isna().sum()

gross                        884
cast_total_facebook_likes      0
dtype: int64

In [22]:
df1= df[['gross', 'cast_total_facebook_likes']]
df1.dropna(inplace= True)

/Users/avidanberman/opt/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  


In [29]:
df1.corr()

gross  cast_total_facebook_likes
gross                      1.0000                     0.2474
cast_total_facebook_likes  0.2474                     1.0000

In [30]:
df1.corr()['gross'][1]

0.24739963797950137

2)
H0) foreign = domestic H1) foreign != domestic

Answer: Two-Sample T-Test

In [92]:
def two_sample_two_tail_ttest(a, b):
    return stats.ttest_ind(a, b, axis=0, equal_var= False, nan_policy='omit')

In [55]:
df2= df[['country', 'gross']]

In [68]:
foreign= df2[df2['country'] != 'USA']['gross']

2       200074175.0
4               NaN
9       301956980.0
12      168368427.0
20      255108370.0
           ...     
5027       673780.0
5028            NaN
5029        94596.0
5034        70071.0
5038            NaN
Name: gross, Length: 1236, dtype: float64

In [69]:
domestic= df2[df2['country'] == 'USA']['gross']

In [74]:
stats.ttest_ind(foreign, domestic, axis=0, equal_var=False, nan_policy='omit')

Ttest_indResult(statistic=-43.77955911028703, pvalue=0.0)

Reject Null Hypothesis as pvalue < .05

3.

H0) 40% of movies = Rated R H1) 40% of movies != Rated R

In [75]:
dfr= df['content_rating'] == 'R'

In [82]:
weightstats.ztest(dfr, value=0, alternative='two-sided', usevar='pooled', ddof=(len(df)-1))

(0.8509421626071682, 0.3948014820818201)

Z-Stat(0.8509421626071682) P-Value(0.3948014820818201) P-Value > .05 Therefore we can't reject the Null hypothesis

4.  (Use Chi Squared)

In [83]:
print(df.keys())

Index(['color', 'director_name', 'num_critic_for_reviews', 'duration',
       'director_facebook_likes', 'actor_3_facebook_likes', 'actor_2_name',
       'actor_1_facebook_likes', 'gross', 'genres', 'actor_1_name',
       'movie_title', 'num_voted_users', 'cast_total_facebook_likes',
       'actor_3_name', 'facenumber_in_poster', 'plot_keywords',
       'movie_imdb_link', 'num_user_for_reviews', 'language', 'country',
       'content_rating', 'budget', 'title_year', 'actor_2_facebook_likes',
       'imdb_score', 'aspect_ratio', 'movie_facebook_likes', 'foreign',
       'domestic'],
      dtype='object')


In [91]:
pd.get_dummies(df['language'], drop_first= True)

Arabic  Aramaic  Bosnian  Cantonese  Chinese  Czech  Danish  Dari  \
0          0        0        0          0        0      0       0     0   
1          0        0        0          0        0      0       0     0   
2          0        0        0          0        0      0       0     0   
3          0        0        0          0        0      0       0     0   
4          0        0        0          0        0      0       0     0   
...      ...      ...      ...        ...      ...    ...     ...   ...   
5038       0        0        0          0        0      0       0     0   
5039       0        0        0          0        0      0       0     0   
5040       0        0        0          0        0      0       0     0   
5041       0        0        0          0        0      0       0     0   
5042       0        0        0          0        0      0       0     0   

      Dutch  Dzongkha  English  Filipino  French  German  Greek  Hebrew  \
0         0         0        1         0       0       0      0       0   
1         0         0        1         0       0       0      0       0   
2         0         0        1         0       0       0      0       0   
3         0         0        1         0       0       0      0       0   
4         0         0        0         0       0       0      0       0   
...     ...       ...      ...       ...     ...     ...    ...     ...   
5038      0         0        1         0       0       0      0       0   
5039      0         0        1         0       0       0      0       0   
5040      0         0        1         0       0       0      0       0   
5041      0         0        1         0       0       0      0       0   
5042      0         0        1         0       0       0      0       0   

      Hindi  Hungarian  Icelandic  Indonesian  Italian  Japanese  Kannada  \
0         0          0          0           0        0         0        0   
1         0          0          0           0        0         0        0   
2         0          0          0           0        0         0        0   
3         0          0          0           0        0         0        0   
4         0          0          0           0        0         0        0   
...     ...        ...        ...         ...      ...       ...      ...   
5038      0          0          0           0        0         0        0   
5039      0          0          0           0        0         0        0   
5040      0          0          0           0        0         0        0   
5041      0          0          0           0        0         0        0   
5042      0          0          0           0        0         0        0   

      Kazakh  Korean  Mandarin  Maya  Mongolian  None  Norwegian  Panjabi  \
0          0       0         0     0          0     0          0        0   
1          0       0         0     0          0     0          0        0   
2          0       0         0     0          0     0          0        0   
3          0       0         0     0          0     0          0        0   
4          0       0         0     0          0     0          0        0   
...      ...     ...       ...   ...        ...   ...        ...      ...   
5038       0       0         0     0          0     0          0        0   
5039       0       0         0     0          0     0          0        0   
5040       0       0         0     0          0     0          0        0   
5041       0       0         0     0          0     0          0        0   
5042       0       0         0     0          0     0          0        0   

      Persian  Polish  Portuguese  Romanian  Russian  Slovenian  Spanish  \
0           0       0           0         0        0          0        0   
1           0       0           0         0        0          0        0   
2           0       0           0         0        0          0        0   
3           0       0           0         0       

color      director_name  num_critic_for_reviews  duration  \
0     Color      James Cameron                   723.0     178.0   
1     Color     Gore Verbinski                   302.0     169.0   
2     Color         Sam Mendes                   602.0     148.0   
3     Color  Christopher Nolan                   813.0     164.0   
4       NaN        Doug Walker                     NaN       NaN   
...     ...                ...                     ...       ...   
5038  Color        Scott Smith                     1.0      87.0   
5039  Color                NaN                    43.0      43.0   
5040  Color   Benjamin Roberds                    13.0      76.0   
5041  Color        Daniel Hsia                    14.0     100.0   
5042  Color           Jon Gunn                    43.0      90.0   

      director_facebook_likes  actor_3_facebook_likes      actor_2_name  \
0                         0.0                   855.0  Joel David Moore   
1                       563.0                  1000.0     Orlando Bloom   
2                         0.0                   161.0      Rory Kinnear   
3                     22000.0                 23000.0    Christian Bale   
4                       131.0                     NaN        Rob Walker   
...                       ...                     ...               ...   
5038                      2.0                   318.0     Daphne Zuniga   
5039                      NaN                   319.0     Valorie Curry   
5040                      0.0                     0.0     Maxwell Moody   
5041                      0.0                   489.0     Daniel Henney   
5042                     16.0                    16.0  Brian Herzlinger   

      actor_1_facebook_likes        gross                           genres  \
0                     1000.0  760505847.0  Action|Adventure|Fantasy|Sci-Fi   
1                    40000.0  309404152.0         Action|Adventure|Fantasy   
2                    11000.0  200074175.0        Action|Adventure|Thriller   
3                    27000.0  448130642.0                  Action|Thriller   
4                      131.0          NaN                      Documentary   
...                      ...          ...                              ...   
5038                   637.0          NaN                     Comedy|Drama   
5039                   841.0          NaN     Crime|Drama|Mystery|Thriller   
5040                     0.0          NaN            Drama|Horror|Thriller   
5041                   946.0      10443.0             Comedy|Drama|Romance   
5042                    86.0      85222.0                      Documentary   

         actor_1_name                                        movie_title  \
0         CCH Pounder                                            Avatar    
1         Johnny Depp          Pirates of the Caribbean: At World's End    
2     Christoph Waltz                                           Spectre    
3           Tom Hardy                             The Dark Knight Rises    
4         Doug Walker  Star Wars: Episode VII - The Force Awakens    ...   
...               ...                                                ...   
5038      Eric Mabius                           Signed Sealed Delivered    
5039      Natalie Zea                         The Following                
5040      Eva Boehnke                              A Plague So Pleasant    
5041        Alan Ruck                                  Shanghai Calling    
5042      John August                                 My Date with Drew    

      num_voted_users  cast_total_facebook_likes          actor_3_name  \
0              886204                       4834             Wes Studi   
1              471220                      48350        Jack Davenport   
2              275868                      11700      Stephanie Sigman   
3             1144337                     106759  Joseph Gordon-Levitt   
4                   8                        143                   N

## Question 3

Now that you have answered all of those questions, the excutive wants you to create a model that predicts the money a movie will make if it is released next year in the US. She wants to use this to evaluate different scripts and then decide which one has the largest revenue potential. Below is a list of potential features you could use in the model. Discuss the steps you would need to take in order to prepare the data for the model. Would you use all of these features in the model? Identify which features you might drop and why, and discuss the process you would use to determine this.  

*Remember you want to be able to use this model to predict the box office gross of a film **before** anyone has seen it.*

- budget: The amount of money spent to make the movie
- title_year: The year the movie first came out in the box office
- years_old: How long has it been since the movie was released
- genre: Each movie is assigned one genre category like action, horror, comedy.
- avg_user_rating: This rating is taken from Rotten tomatoes, and is the average rating given to the movie by the audience. 
- actor_1_facebook_likes: The number of likes that the most popular actor in the movie has
- total_cast_facebook_likes: The sum of likes for the three most popular actors in the movie
- language: the orginal spoken language of the film


In [9]:
#your answer here

## Question 4

Below is the summary output of a fitted OLS model.  Look at the summary of this OLS model and identify any key takeaways from it.  How ‘good’ is this model? Which features help to explain the variance in the target variable? Which do not? 


<img src="ols_summary.png" style="withd:300px;">

In [10]:
#your answer here

## Question 5

**Bayes Theorem**

An advertising executive is studying television viewing habits of married men and women during prime time hours. Based on the past viewing records he has determined that during prime time wives are watching television 60% of the time. It has also been determined that when the wife is watching television, 40% of the time the husband is also watching. When the wife is not watching the television, 30% of the time the husband is watching the television. Find the probability that if the husband is watching the television, the wife is also watching the television.

In [11]:
#your answer here